FirebaseRemoteConfig/Sources/FIRRemoteConfig.m

```diff
@@ -237,6 +240,61 @@ - (void)callListeners:(NSString *)key config:(NSDictionary *)config {
   }
 }
 
+- (void)setCustomSignals:(nullable NSDictionary<NSString *, NSObject *> *)customSignals
+          WithCompletion:(void (^_Nullable)(NSError *_Nullable error))completionHandler {
+  void (^setCustomSignalsBlock)(void) = ^{
+    if (!customSignals) {
+      if (completionHandler) {
+        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Any reason we have to dispatch the completion block back to main thread? Ideally if SDK work does not involve UI we should try stay away from main as much as possible.

Reply:
Thanks for bringing this up! I have updated the code to use dispatch_get_global_queue to address your concern. This ensures we don't unnecessarily block the main thread in this case.

Reply:
I actually wonder why setting custom signals would need a completion handler? To me it feels like a `fetch` setting, the signals influence the config returned from fetch which then may or may not update the UI, but just setting signals shouldn't have any UI impact?

Reply:
As finalized in the Custom Targeting Review discussion sync, since this change isn't UI-related, we'll avoid dispatching the completion block to the main queue. dispatch_get_global_queue is a good choice here to ensure we don't unnecessarily block the main thread.